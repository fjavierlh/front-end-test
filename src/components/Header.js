import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

export const Header = ({ logoUrl, title, cartCount }) => {
  return (
    <HeaderStyled>
      <Link to="/" className="brand">
        {logoUrl ? (
          <img className="logo" src={logoUrl} alt="logo"></img>
        ) : (
          <h4>{title}</h4>
        )}
      </Link>
      <div className="cart">
        <p>Items in your cart</p>
        <div className="cart-count">{cartCount}</div>
      </div>
    </HeaderStyled>
  );
};

const HeaderStyled = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
  & > .logo {
    width: 200px;
    max-width: 200px;
  }
  & > .brand {
    text-decoration: none;
    color: #333;
  }
  & > .cart {
    display: flex;
    flex-wrap: wrap;
    gap: 0.5em;
    text-transform: uppercase;
    align-items: center;
    font-size: 0.9em;
    font-weight: bold;
  }
  & > .cart > .cart-count {
    display: flex;
    margin-right: 1rem;
    justify-content: center;
    align-items: center;
    background-color: #333;
    color: #fff;
    height: 1.5rem;
    width: 1.5rem;
    font-weight: bold;
    font-size: 0.8rem;
    border-radius: 100%;
  }
`;

export default Header;
