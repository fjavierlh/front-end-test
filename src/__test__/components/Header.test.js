import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from '../../components/Header';

describe('Header', () => {
  test('should render Header component', () => {
    const brandTitle = 'Mobile devices shop';
    render(
      <BrowserRouter>
        <Header title={brandTitle} />
      </BrowserRouter>
    );
    const brand = screen.getByText(/mobile devices shop/i);
    expect(brand).toBeInTheDocument();
  });

  test('should render Header component with cart count with 0 default value', () => {
    const brandTitle = 'Mobile devices shop';
    render(
      <BrowserRouter>
        <Header title={brandTitle} />
      </BrowserRouter>
    );
    const cartCount = screen.getByText(/0/i);
    expect(cartCount).toBeInTheDocument();
  });

  test('should render Header component with cart count passed by props', () => {
    const brandTitle = 'Mobile devices shop';
    render(
      <BrowserRouter>
        <Header title={brandTitle} cartCount={10} />
      </BrowserRouter>
    );
    const cartCount = screen.getByText(/10/i);
    expect(cartCount).toBeInTheDocument();
  });

  test('should match snapshot with cart count passed by props', () => {
    const brandTitle = 'Mobile devices shop';
    const { container } = render(
      <BrowserRouter>
        <Header title={brandTitle} cartCount={10} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('should match snapshot with cart count default value by props', () => {
    const brandTitle = 'Mobile devices shop';
    const { container } = render(
      <BrowserRouter>
        <Header title={brandTitle} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
