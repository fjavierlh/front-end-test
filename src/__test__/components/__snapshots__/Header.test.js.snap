// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot with cart count default value by props 1`] = `
<div>
  <header
    class="sc-bdvvtL cKpnYO"
  >
    <a
      class="brand"
      href="/"
    >
      <h4>
        Mobile devices shop
      </h4>
    </a>
    <div
      class="cart"
    >
      <p>
        Items in your cart
      </p>
      <div
        class="cart-count"
      >
        0
      </div>
    </div>
  </header>
</div>
`;

exports[`Header should match snapshot with cart count passed by props 1`] = `
<div>
  <header
    class="sc-bdvvtL cKpnYO"
  >
    <a
      class="brand"
      href="/"
    >
      <h4>
        Mobile devices shop
      </h4>
    </a>
    <div
      class="cart"
    >
      <p>
        Items in your cart
      </p>
      <div
        class="cart-count"
      >
        10
      </div>
    </div>
  </header>
</div>
`;
